File: go/clock/clock.go
package clock

import "fmt"

const testVersion = 4

type Clock struct {
	Hour   int
	Minute int
}

func totalTimeCalc(hour, min int) (int, int) {
	hours := min / 60
	minutes := min % 60

	hour += hours
	hour %= 24

	if minutes < 0 {
		minutes += 60
		hour--
	}

	if hour < 0 {
		hour += 24
	}
	return hour, minutes
}

func New(hour, minute int) Clock {
	hour, minute = totalTimeCalc(hour, minute)
	return Clock{hour, minute}
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.Hour, c.Minute)
}

func (c Clock) Add(minutes int) Clock {
	hour, minute := c.Hour, (c.Minute + minutes)
	return New(totalTimeCalc(hour, minute))
}

File: javascript/clock/clock.js
var Clock = function(hour, min) {
    min = (typeof min !== 'undefined') ? min : 0;
    d = new Date()
    d.setUTCHours(hour)
    d.setUTCMinutes(min)
    this.time = d
};

Clock.prototype.equals = function(clock) {
    return this.toString() === clock.toString();
};

Clock.prototype.toString = function() {
    return JSON.stringify(this.time).slice(12, 17)
}

Clock.prototype.plus = function(minutes) {
    current = this.time.getMinutes()
    this.time.setUTCMinutes(current + minutes)
    return this.toString()
}

Clock.prototype.minus = function(minutes) {
    current = this.time.getMinutes()
    this.time.setUTCMinutes(current - minutes)
    return this.toString()
}

var at = function(hour, min) {
    clock = new Clock(hour, min)
    return clock
};


var at = module.exports = { at };